src/main/java/org/polypheny/jdbc/meta/PolyphenyDatabaseMetadata.java

```diff
@@ -0,0 +1,1367 @@
+/*
+ * Copyright 2019-2024 The Polypheny Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.polypheny.jdbc.meta;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.RowIdLifetime;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.polypheny.db.protointerface.proto.Column;
+import org.polypheny.db.protointerface.proto.DbmsVersionResponse;
+import org.polypheny.db.protointerface.proto.Entity;
+import org.polypheny.db.protointerface.proto.ForeignKey;
+import org.polypheny.db.protointerface.proto.Function;
+import org.polypheny.db.protointerface.proto.Index;
+import org.polypheny.db.protointerface.proto.Namespace;
+import org.polypheny.db.protointerface.proto.PrimaryKey;
+import org.polypheny.db.protointerface.proto.Procedure;
+import org.polypheny.db.protointerface.proto.Table;
+import org.polypheny.db.protointerface.proto.TableType;
+import org.polypheny.db.protointerface.proto.Type;
+import org.polypheny.jdbc.ConnectionString;
+import org.polypheny.jdbc.PolyConnection;
+import org.polypheny.jdbc.PrismInterfaceClient;
+import org.polypheny.jdbc.PrismInterfaceErrors;
+import org.polypheny.jdbc.PrismInterfaceServiceException;
+import org.polypheny.jdbc.properties.DriverProperties;
+import org.polypheny.jdbc.properties.PropertyUtils;
+
+public class PolyphenyDatabaseMetadata implements DatabaseMetaData {
+
+    private static final int NO_VERSION = -1;
+    private ConnectionString connectionString;
+
+    private NullSorting nullSorting;
+
+    private PrismInterfaceClient prismInterfaceClient;
+
+    private PolyConnection polyConnection;
+
+    private String productName;
+    private String productVersion;
+    private int databaseMinorVersion = NO_VERSION;
+    private int databaseMajorVersion = NO_VERSION;
+
+
+    // TODO TH: remove and hardcode this
+    private enum NullSorting {
+        START,
+        END,
+        HIGH,
+        LOW
+    }
+
+
+    public PolyphenyDatabaseMetadata( PrismInterfaceClient prismInterfaceClient, ConnectionString target ) {
+        this.prismInterfaceClient = prismInterfaceClient;
+        this.connectionString = target;
+        //TODO TH: check what polypheny does...
```

Comment:
Todo?

Reply:
I just tested this. The proper setting is "NullSorting.END".

Reply:
Can we add a test case for this to ensure this update in case the behavior of Polypheny changes in future?

Reply:
Just added some tests for this.